src/icrar/leap-accelerate/ms/MeasurementSet.cc

```diff
@@ -0,0 +1,116 @@
+/**
+*    ICRAR - International Centre for Radio Astronomy Research
+*    (c) UWA - The University of Western Australia
+*    Copyright by UWA (in the framework of the ICRAR)
+*    All rights reserved
+*
+*    This library is free software; you can redistribute it and/or
+*    modify it under the terms of the GNU Lesser General Public
+*    License as published by the Free Software Foundation; either
+*    version 2.1 of the License, or (at your option) any later version.
+*
+*    This library is distributed in the hope that it will be useful,
+*    but WITHOUT ANY WARRANTY; without even the implied warranty of
+*    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+*    Lesser General Public License for more details.
+*
+*    You should have received a copy of the GNU Lesser General Public
+*    License along with this library; if not, write to the Free Software
+*    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+*    MA 02111-1307  USA
+*/
+
+#include "MeasurementSet.h"
+#include <icrar/leap-accelerate/ms/utils.h>
+
+namespace icrar
+{
+    MeasurementSet::MeasurementSet(std::string filepath)
+    {
+        m_measurementSet = std::make_unique<casacore::MeasurementSet>(filepath);
+        m_msmc = std::make_unique<casacore::MSMainColumns>(*m_measurementSet);
+        m_msc = std::make_unique<casacore::MSColumns>(*m_measurementSet);
+    }
+
+    MeasurementSet::MeasurementSet(const casacore::MeasurementSet& ms)
+    {
+        m_measurementSet = std::make_unique<casacore::MeasurementSet>(ms);
+        m_msmc = std::make_unique<casacore::MSMainColumns>(*m_measurementSet);
+        m_msc = std::make_unique<casacore::MSColumns>(*m_measurementSet);
+    }
+
+    MeasurementSet::MeasurementSet(std::istream& stream)
+    {
+        // don't skip the whitespace while reading
+        std::cin >> std::noskipws;
+
+        // use stream iterators to copy the stream to a string
+        std::istream_iterator<char> it(std::cin);
+        std::istream_iterator<char> end;
+        std::string results = std::string(it, end);
+    }
+
+    unsigned int MeasurementSet::GetNumStations() const
+    {
+        return m_measurementSet->antenna().nrow();
+    }
+
+    unsigned int MeasurementSet::GetNumPols() const
+    {
+        if(m_measurementSet->polarization().nrow() > 0)
+        {
+            return m_msc->polarization().numCorr().get(0);
+        }
+        else
+        {
+            throw icrar::not_implemented_exception(__FILE__, __LINE__);
+        }
+        
+    }
+
+    unsigned int MeasurementSet::GetNumBaselines() const
+    {
+        const size_t num_stations = (size_t)GetNumStations();
+        return num_stations * (num_stations + 1) / 2; //TODO: +/- 1???
```

Comment:
I think this is related to our discussion via email. This currently gives you the total number of baselines that are possible with that many stations (and autocorrelation), but doesn't necessarily reflect the number of visibilities for which there is data on the MS. No need to act on it in this PR, just wanted to point out the connection.